### ISSUE TITLE
[Epic] Sharded persist reads/writes

### ISSUE THREAD

**Issue by sploiselle** [MaterializeInc/materialize#13884]
### Feature request

Currently, Materialize only uses a single Timely worker to read from persist (`persist_source`) and write to it (`persist_sink`). This leaves a lot of throughput on the table by not taking advantage of the available parallelism.

## Reads

Tracked in #12865

tl;dr: Sharding reads from persist immediately achievable, and we have a prototype working.

- [x] Design (in Notion under name "Sharded reads"
- [x] Prototype #13849
- [x] Tech lead/area owner review  #13849
- [ ] Benchmarking (maybe parallelized with writes work)
- [x] PR (#13849)

## Writes

Tracked in #13886

tl;dr Sharding writes to persist seems achievable, but the design space has not yet been explored.

- [x] Design
- [x] Prototype
- [x] Tech lead/area owner review
- [ ] Benchmarking
- [x] PR (#14001)

**Comment by nmeagan11** [comment]
Now that #14001 has merged, is this ready for QA, or do we want to do some benchmarking first? cc @aljoscha

**Comment by aljoscha** [comment]
I'd say it's ready for QA

**Comment by philip-stoev** [comment]
QA blocked by https://github.com/MaterializeInc/materialize/issues/15300
